src/snowflake/snowpark/modin/plugin/extensions/dataframe_overrides.py

```diff
@@ -7,23 +7,1474 @@
+    escape=True,
+    notebook=False,
+    border=None,
+    table_id=None,
+    render_links=False,
+    encoding=None,
+):  # noqa: PR01, RT01, D200
+    pass
+
+
+@register_dataframe_not_implemented()
+def to_parquet(
+    self,
+    path=None,
+    engine="auto",
+    compression="snappy",
+    index=None,
+    partition_cols=None,
+    storage_options: StorageOptions = None,
+    **kwargs,
+):
+    pass
+
+
+@register_dataframe_not_implemented()
+def to_period(
+    self, freq=None, axis=0, copy=True
+):  # pragma: no cover # noqa: PR01, RT01, D200
+    pass
+
+
+@register_dataframe_not_implemented()
+def to_records(
+    self, index=True, column_dtypes=None, index_dtypes=None
+):  # noqa: PR01, RT01, D200
+    pass
+
+
+@register_dataframe_not_implemented()
+def to_stata(
+    self,
+    path: FilePath | WriteBuffer[bytes],
+    convert_dates: dict[Hashable, str] | None = None,
+    write_index: bool = True,
+    byteorder: str | None = None,
+    time_stamp: datetime.datetime | None = None,
+    data_label: str | None = None,
+    variable_labels: dict[Hashable, str] | None = None,
+    version: int | None = 114,
+    convert_strl: Sequence[Hashable] | None = None,
+    compression: CompressionOptions = "infer",
+    storage_options: StorageOptions = None,
+    *,
+    value_labels: dict[Hashable, dict[float | int, str]] | None = None,
+):
+    pass
+
+
+@register_dataframe_not_implemented()
+def to_xml(
+    self,
+    path_or_buffer=None,
+    index=True,
+    root_name="data",
+    row_name="row",
+    na_rep=None,
+    attr_cols=None,
+    elem_cols=None,
+    namespaces=None,
+    prefix=None,
+    encoding="utf-8",
+    xml_declaration=True,
+    pretty_print=True,
+    parser="lxml",
+    stylesheet=None,
+    compression="infer",
+    storage_options=None,
+):
+    pass
+
+
+@register_dataframe_not_implemented()
+def __delitem__(self, key):
+    pass
+
+
+@register_dataframe_accessor("attrs")
+@dataframe_not_implemented()
+@property
+def attrs(self):  # noqa: RT01, D200
+    pass
+
+
+@register_dataframe_accessor("style")
+@dataframe_not_implemented()
+@property
+def style(self):  # noqa: RT01, D200
+    pass
+
+
+@register_dataframe_not_implemented()
+def __reduce__(self):
+    pass
+
+
+@register_dataframe_not_implemented()
+def __divmod__(self, other):
+    pass
+
+
+@register_dataframe_not_implemented()
+def __rdivmod__(self, other):
+    pass
+
+
+# The from_dict and from_records accessors are class methods and cannot be overridden via the
+# extensions module, as they need to be foisted onto the namespace directly because they are not
+# routed through getattr. To this end, we manually set DataFrame.from_dict to our new method.
+@snowpark_pandas_telemetry_method_decorator
+@dataframe_not_implemented()
+def from_dict(
+    cls, data, orient="columns", dtype=None, columns=None
+):  # pragma: no cover # noqa: PR01, RT01, D200
+    pass
+
+
+DataFrame.from_dict = from_dict
+
+
+@snowpark_pandas_telemetry_method_decorator
```

Comment:
is this decorator required for all override methods?

Reply:
No, but it is required for most. We don't record telemetry for many methods prefixed by an underscore (like `_to_datetime`).

Reply:
Can we have a unit test to check all public api in dataframe_overrides or series_override have telemetry decorator covered?

Reply:
Is there a good way to check this in unit tests? I don't think there's an easy way to see if a function has a telemetry annotation or not, without actually running the function and checking the logged telemetry.